Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs

```diff
@@ -756,21 +688,15 @@ protected override void OnSetState(SerializationInfo info, StateMode mode)
         _stateStack.Clear();
         if (info.Values.ContainsKey("_stateStack"))
         {
-          //string xml = info.GetValue<string>("_stateStack");
-          byte[] xml = info.GetValue<byte[]>("_stateStack");
-          MobileList<SerializationInfo> list = (MobileList<SerializationInfo>)MobileFormatter.Deserialize(xml);
-          SerializationInfo[] layers = list.ToArray();
-          Array.Reverse(layers);
-          foreach (SerializationInfo layer in layers)
-            _stateStack.Push(layer);
+          var stackArray = info.GetValue<byte[][]>("_stateStack");
+          _stateStack.Clear();
```

Comment:
_stateStack.Clear(); is being called twice in the same control flow. Once just before the **if** statement and once inside it. It looks like this is probably just a copy/past mistake from the implementation in UndoableBase.

I'm not sure if the call before the **if** statement or the one inside is more appropriate.

Reply:
Thanks, will be fixed in another PR